sycl/test-e2e/Matrix/joint_matrix_apply_two_matrices_impl.hpp

```diff
@@ -0,0 +1,147 @@
+//==------- joint_matrix_apply_two_matrices_impl.hpp  - DPC++
+// joint_matrix---==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+template <typename Tc, typename Ta, size_t M, size_t N>
+bool assert_ops_ref(Tc *C, Tc *D, Ta *A, Ta *Ar) {
+  for (size_t i = 0; i < M; i++)
+    for (size_t j = 0; j < N; j++) {
+      Tc diffc = D[i * N + j] - C[i * N + j] * 2;
+      Ta diffa = Ar[i * N + j] - (A[i * N + j] + 42);
+      if constexpr (std::is_same_v<Ta, bfloat16>) {
+        if (std::fabs(diffc) > FLOAT_EPSILON ||
+            std::fabs(diffa) > FLOAT_EPSILON) {
+          return false;
+        } else {
+          if (std::fabs(diffc) > 0 || std::fabs(diffa) > 0) {
+            return false;
+          }
+        }
+      }
+    }
+  return true;
+}
+template <typename Tc, typename Ta, size_t TM, size_t TN, size_t TK, size_t M,
+          size_t N, class kernel_name>
+bool apply_two_matrices(Tc *C, Tc *D, Ta *A, Ta *Ar, queue q) {
+  size_t NDRangeM = M / TM;
+  size_t NDRangeN = N / TN;
+
+  size_t sg_size = get_sg_size<kernel_name>(q);
+  q.submit([&](handler &cgh) {
+     cgh.parallel_for<kernel_name>(
+         nd_range<2>({NDRangeM, NDRangeN * sg_size}, {1, 1 * sg_size}),
+         [=](nd_item<2> spmd_item)
+#ifdef SG_SZ
```

Comment:
currently I don't see test for sg=32, so I suggest either remove this code or add test for sg=32.